src/sqlfluff/cli/commands.py

```diff
@@ -575,6 +576,28 @@ def lint(
         sys.exit(0)
 
 
+def _mark_failed_files_unfixable(lint_result: LintingResult) -> int:
+    """Discard lint fixes for files with templating or parse errors.
+
+    Returns 1 if there are any files with templating or parse errors after
+    filtering, else 0. (Intended as a process exit code.)
+    """
+    total_errors, num_filtered_errors = lint_result.mark_failed_files_unfixable()
```

Comment:
One of them is a wrapper around the other, and they're in different modules. There's a similar (albeit exact) duplication of function names across the linter modules and classes, because of the layered structure of the linter (commands -> LintingResult -> LintedDir -> LintedFile:
* `get_violations()`
* `num_violations()`
* `persist_changes()`
* etc.

Reply:
Renamed both these functions. The names are more distinct now.